internal/jobworker/cgroup/cgroup_test.go

```diff
@@ -0,0 +1,228 @@
+package cgroup
+
+import (
+	"bufio"
+	"errors"
+	"io/fs"
+	"os"
+	"os/exec"
+	"path"
+	"reflect"
+	"testing"
+)
+
+func TestCleanup(t *testing.T) {
+	if !isRoot() {
```

Comment:
If the mount path of the `Service` was changed to  `t.TempDir()` would all these tests still need to run as root? You would also get the added benefit or not leaving any artifacts around in the hosts cgroups directory in the event there are issues with tests.

Reply:
I've updated the tests and the Service to be able to handle any mount path set.

Reply:
+1 for Tim's suggestion, cgroups is mostly writing files, the tests shouldn't care about root or need special permissions.

Reply:
I've left the tests that require root, but I've added a test, `TestControllers`, that I think is providing the testing the two of you are looking for.